MailCore/Cache/MailboxManager/MailboxManager.swift

```diff
@@ -180,26 +180,28 @@ public final class MailboxManager: ObservableObject, MailboxManageable {
     }
 
     func keepCacheAttributes(
-        for message: Message,
+        forLiveMessage liveMessage: Message,
         keepProperties: MessagePropertiesOptions,
         using realm: Realm
     ) {
-        guard let savedMessage = realm.object(ofType: Message.self, forPrimaryKey: message.uid) else {
+        guard let savedMessage = realm.object(ofType: Message.self, forPrimaryKey: liveMessage.uid) else {
             return
         }
-        message.inTrash = savedMessage.inTrash
+        liveMessage.inTrash = savedMessage.inTrash
         if keepProperties.contains(.fullyDownloaded) {
-            message.fullyDownloaded = savedMessage.fullyDownloaded
+            liveMessage.fullyDownloaded = savedMessage.fullyDownloaded
         }
         if keepProperties.contains(.body), let body = savedMessage.body {
-            message.body = Body(value: body)
+            liveMessage.body = Body(value: body).detached()
         }
         if keepProperties.contains(.localSafeDisplay) {
-            message.localSafeDisplay = savedMessage.localSafeDisplay
+            liveMessage.localSafeDisplay = savedMessage.localSafeDisplay
         }
         if keepProperties.contains(.attachments) {
             let attachments = savedMessage.attachments.map { Attachment(value: $0.freeze()) }
```

Comment:
I know you didn't touch this line but it would make more sense to use detached. Freeze isn't useful in this case.
```suggestion
            let attachments = savedMessage.attachments.map { $0.detached() }
```

Reply:
Sure

Also I'll keep an eye for this since it exists elsewhere in the app. We blindly use this `Type(value: )` at other locations